# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1469454568 14400
#      Mon Jul 25 09:49:28 2016 -0400
# Node ID 2258495e864a64d936d9c421a2fcc428cf273ad3
# Parent  8358b9ef5def18c1d37837accb5ad21a302a0a20
* mk-doc-cache.pl: Fix substitutions to escape block comment markers.

diff --git a/doc/interpreter/mk-doc-cache.pl b/doc/interpreter/mk-doc-cache.pl
--- a/doc/interpreter/mk-doc-cache.pl
+++ b/doc/interpreter/mk-doc-cache.pl
@@ -26,16 +26,18 @@ use File::Temp qw(tempfile);
 
 die "usage: mk_doc_cache OUTPUT-FILE SRCDIR MACRO-FILE ... -- DOCSTRINGS-FILE ..." if (@ARGV < 3);
 
 $makeinfo_command = "makeinfo --no-headers --no-warn --force --no-validate --fill-column=1024";
 
 $output_file = shift (@ARGV);
 $top_srcdir = shift (@ARGV);
 
+$amp = "@";
+
 ## Constant patterns.
 
 $doc_delim = "\x{1d}";
 $doc_delim_pat = qr/^\x{1d}/;
 $tex_delim_pat = qr/\Q-*- texinfo -*-\E/;
 $private_name_pat = qr/^__.+__/;
 
 $text = "";
@@ -78,25 +80,26 @@ foreach $arg (@ARGV)
             }
 
           next if ($in_header);
 
           next if (/$tex_delim_pat/);
 
           ## Escapes for symbol names.
 
-          s/$doc_delim_pat(([#%]|)[{}]|@)/$doc_delim@$1/;
+          s/$doc_delim_pat([{}@])/$doc_delim$amp$1/;
+          s/$doc_delim_pat([#%])([{}])/$doc_delim$1$amp$2/;
           $text .= $_;
         }
     }
 }
 
 $text .= $doc_delim;
 
-($fh, $file) = tempfile (UNLINK => 1);
+($fh, $file) = tempfile ();
 print $fh "$text";
 close ($fh);
 
 $cmd = "$makeinfo_command $file";
 open (CMD, "-|", $cmd) or die "Unable to start makeinfo command $cmd\n";
 $formatted_text = "";
 while (<CMD>)
 {
